parking_lot.rb

```diff
@@ -0,0 +1,20 @@
+require_relative "./car.rb"
+
+class ParkingLot
+  def initialize (filename)
+    @cars_hash = {}
+    cars = File.open(filename, "r")
+    cars_array = cars.read.split("\n\n\n").map{|words| words.split}.map{|car| @cars_hash[car[0]] = car[1]}
+    puts @cars_hash
+  end 
+
+  def list_cars
+    @cars_hash.each do |key, value|
+      key = Car.new(key, value)
```

Comment:
Once you extract Car.new to hash builder, now you just have to call .what_car here and can make it a one liner!